class-00/lab-b/lab/app.js

```diff
@@ -0,0 +1,208 @@
+// // The way an object is returned is different with an arrow function, too.
+// // Here is how we return an object without arrow functions
+// const oldObject = function(array) {
+//   return {
+//     firstValue: array[0],
+//     secondValue: array[1],
+//     thirdValue: array[2]
+//   };
+// };
+
+// // TODO: Uncomment the following line of code to see the output in the browser console
+// console.log('Hello from the old object function', oldObject(['hi', 'hello', 'are you there?']));
+
+
+// // STEP 9
+// // With an arrow function, we need to wrap our object in parentheses
+// // Otherwise, it will be interpreted as a code block
+// const newObject = array => ({
+//   firstValue: array[0],
+//   secondValue: array[1],
+//   thirdValue: array[2]
+// });
+
+// // TODO: Uncomment the following line of code to see the output in the browser console
+// console.log('Hello from the new object function', newObject(['hi', 'hello', 'are you there?']));
+
+
+// // STEP 10
+// // Uncomment the console.log lines to view the output of each function in the browser console.
+// // Refactor each function into an arrow function.
+// // Write your solutions on a single line wherever possible.
+
+
+
+const sum = (a, b, c, d) => a + b + c + d;
+
+// TODO: Uncomment the following line of code to see the output in the browser console
+console.log(sum(1, 2, 3, 4));
+
+
+let objectLit = () =>({
+  key1: 'value1',
+  key2: 'value2',
+  key3: 'value3',
+});
+
+// TODO: Uncomment the following line of code to see the output in the browser console
+console.log(objectLit());
+
+
+let sumAndProduct = (a, b) => {
+  let sum = a + b;
+  let product=  a * b;
+  return [sum, product];
+};
+
+// TODO: Uncomment the following line of code to see the output in the browser console
+console.log(sumAndProduct(3, 9));
+
+
+let message = (name) =>{
+  return `Hello, ${name}!`;
+};
+
+// TODO: Uncomment the following line of code to see the output in the browser console
+// let Student = (name, age, hometown) => {
+//   this.name = name;
+//   this.age = age;
+//   this.hometown = hometown;
+// };
+
+// let joe = new Student('Joe', 'Schmoe', 100);
+
+
+let Student = function(name, age, hometown) {
+  this.name = name;
+  this.age = age;
+  this.hometown = hometown;
+};
+
+let joe = new Student('Joe', 'Schmoe', 100);
+
+// TODO: Uncomment the following line of code to see the output in the browser console
+// Note that the arrow function will cause this code to break!
+console.log(joe);
+
+// TODO: After viewing the previous console.log(), return the code to a working state.
+
+
+
+Student.prototype.greeting = function() {
+  return `Hi, my name is ${this.name}`;
+};
+
+// TODO: Uncomment the following line of code to see the output in the browser console
+// Note that the arrow function will cause this method to break!
+console.log(joe.greeting());
+
+// TODO: After viewing the previous console.log(), return the code to a working state.
+
+
+
+Student.courseName = () =>{

```

Comment:
Same as above. This can be refactored into a one-liner like this:
`Student.courseName = () => 'This student is enrolled in Code 301.';`